src/main/java/org/example/servletstexts/TextsServlet.java

```diff
@@ -3,111 +3,128 @@
-                        respWriter.write("Invalid format");
-                    }
+                if (pathParts.length == 2 && !pathParts[1].isEmpty()) {
+                    deleteTextById(pathParts[1], resp);
                 } else {
-                    resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                    respWriter.write("Invalid request path. You entered an extra '/'");
+                    throw new BadRequestException(INVALID_REQUEST_PATH + " You entered an extra '/'");
                 }
             }
         } catch (IOException e) {
-            resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
-            resp.getWriter().write("Internal server error");
+            throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
         }
     }
 
     @Override
-    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    protected void handlePost(HttpServletRequest req, HttpServletResponse resp) {
         resp.setContentType("application/json");
 
-        try (PrintWriter respWriter = resp.getWriter()) {
+        try {
+            PrintWriter respWriter = resp.getWriter();
             ObjectNode requestBody = objectMapper.readValue(req.getReader(), ObjectNode.class);
             String text = requestBody.get("text").asText();
 
             if (text != null && !text.trim().isEmpty()) {
                 texts.put(currentId, text);
+
                 ObjectNode responseBody = objectMapper.createObjectNode();
                 responseBody.put("text", text);
                 responseBody.put("message", "Text saved with id: " + currentId);
                 responseBody.put("textId", currentId);
                 respWriter.write(responseBody.toString());
+
                 currentId++;
             } else {
-                resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                respWriter.write("Text cannot be empty");
+                throw new BadRequestException(EMPTY_TEXT_ERROR);
             }
+
         } catch (IOException e) {
-            resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
-            resp.getWriter().write("Internal server error");
+            throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
+        }
+    }
+
+
+    private void deleteTextById(String id, HttpServletResponse resp) throws IOException {
+        try {
+            int textId = Integer.parseInt(id);
+            if (texts.remove(textId) != null) {
+                resp.getWriter().write("Text with id " + textId + " has been deleted");
+            } else {
+                throw new NotFoundException(TEXT_NOT_FOUND + " with id " + textId);
+            }
+        } catch (NumberFormatException e) {
+            throw new BadRequestException(INVALID_FORMAT);
         }
     }
 
+    private void getTextById(String id, HttpServletResponse resp) throws IOException {
```

Comment:
if method starts with  `get`  then it supposed to return something. This method return void. Rename method or (what will be better) split your logic into String getText(String id) and other logic of how text send to client write ion handleGet method.